### ISSUE TITLE
Can't Install Signal Private Messenger

### ISSUE THREAD

**Issue by freeasinbear** [Noitidart/Chrome-Store-Foxified#18]
The process seems to get stuck at "Signing "Signal Private Messenger"... Submitting request to addons.mozilla.org (AMO)" and at the end I get a red bar. When I click on "Show failed JSON" I get: 

`{
    "msg": "signing-failed: submit failed",
    "xhr": {
        "status": 401,
        "response": {
            "detail": "Signature has expired."
        }
    }
}`

This app shouldn't be really bounded to chrome, it's more like a standalone that installs itself through Chrome as an extension, so I guess it shouldn't be a problem to add It on Firefox

**Comment by Noitidart** [comment]
Thanks for this report. Can you please provide me a link to the extension?

**Comment by fabriziodicandia** [comment]
Here it is: https://chrome.google.com/webstore/detail/signal-private-messenger/bikioccmkafdpakkkcpdbppfkghcmihk

**Comment by freeasinbear** [comment]
@Juesto Has it been solved?

**Comment by thrdroom** [comment]
I can confirm. I dont have the same but a similar problem trying to installing signal [https://github.com/Noitidart/Chrome-Store-Foxified/issues/46](https://github.com/Noitidart/Chrome-Store-Foxified/issues/46)

**Comment by Noitidart** [comment]
A patch for signing this addon landed here - https://github.com/mozilla/addons-server/issues/2802#issuecomment-222561801

However in my test today it seems to still happen. So it might be their patch has not yet reached production.

**Comment by Noitidart** [comment]
I got around the signing issue and I see that this extension uses unsupported APIs. This is the permissions requested by this addon in the manifest.json:

    "permissions": ["unlimitedStorage", "notifications", {
        "fileSystem": ["write"]
    }, "alarms", "fullscreen", "audioCapture"],

`unlimitedStorage`, `audioCapture` and `fileSystem` are all not supported - https://developer.mozilla.org/en-US/Add-ons/WebExtensions/manifest.json/permissions#permissions

**Comment by Noitidart** [comment]
After getting around the signing issue, the extension "Signal" fails to install because it uses incompatible stuff:

> 1476041036769	addons.webextension.<unknown>	ERROR	Loading extension 'null': Reading manifest: Error processing permissions.2: Expected string instead of {"fileSystem":["write"]}  Log.jsm:753

> Invalid match pattern: '[object Object]'  MatchPattern.jsm:52

It seems Firefox WebExtensions API does not support this API. It's not even listed here - https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Chrome_incompatibilities

**Comment by Noitidart** [comment]
I filed a bug on bugzilla:

https://bugzilla.mozilla.org/show_bug.cgi?id=1308796

Once they fix it, this addon should install fine, but some of the APIs won't work.

**Comment by Noitidart** [comment]
They resolved the bug, apparently "Signal" is not a extension, it is an "App". "App"s are not supported by the WebExtension API. Per this comment here - https://bugzilla.mozilla.org/show_bug.cgi?id=1308796#c1

@thrdroom, @freeasinbear is this true, it is an app? Not an extension?

**Comment by Noitidart** [comment]
This is also something that was shared with me on IRC:

>  14:29:54 	<aswan>	noit-away: fileSystem and audioCapture are both for chrome apps, not extensions

> 14:30:39 	<aswan>	as for unlimitedStorage, there are several facets to it, but we don't actually put any quote on browser.storage.local right now so you don't need it for that

> 14:30:46 	<aswan>	if you want it for indexedDB or something, there is an open bug

**Comment by freeasinbear** [comment]
> They resolved the bug, apparently "Signal" is not a extension, it is an "App". "App"s are not supported by the WebExtension API. Per this comment here - https://bugzilla.mozilla.org/show_bug.cgi?id=1308796#c1

@Noitidart Yeah that's right, it is listed as an app, so there's nothing that can be done?

**Comment by Noitidart** [comment]
@freeasinbear - from our end we can get into the code of Signal and transition it to an extension. If we don't want to do that, we can wait to see if the WebExt team adds app supprort.

**Comment by thrdroom** [comment]
@Noitidart I hope for the support of apps by the WebExt team as i dont have much spare time and not that knowledged about the development of extensions.

My deepest hope is that signal messenger gets official supported and developed for firefox by @moxie0

**Comment by Noitidart** [comment]
Hey all I reached out to the Signal team asking if they can give us a universal or firefox specific client - I'll let you know what their reply is.
